### DISCUSSION TITLE
How to send ESC from js script to switch to VIM normal mode?

### DISCUSSION THREAD

**Discussion by kkarsten62** [ajaxorg/ace#5463] [Q&A] [answered]
Hi Ace community,
I am just started to implement the Ace editor in my webpage. So far so good, all works fine. I prefer to use VIM for keybinding.
In a javascript method I need to make sure that the VIM mode is always in normal mode so a switch from insert to normal mode is needed. Interactively in VIM this will done with key ESC or Ctrl-C.
So question is: How to send ESC to the ace editor via a js? Or is there another tip to bring VIM to normal mode from a js script?

Thanks for a helping hand.
Karl ...

**Comment by nightwing** [comment]
you can do `require("ace/keyboard/vim").Vim.exitInsertMode(editor.state.cm)` or `editor.onCommandKey({}, 0, 27)`

**Comment by jonasermert** [comment, marked as answer]
```
var editor = ace.edit("your-editor-id"); // replace 'your-editor-id' with the actual ID of your Ace editor instance

// Switch to normal mode programmatically
var event = new KeyboardEvent('keypress', {
  keyCode: 27,  // Key code for ESC key
  which: 27     // Additional property for some browsers
});

editor.textInput.getElement().dispatchEvent(event);

```

**Comment by kkarsten62** [comment]
Thanks for helping me ...  :-)
I implemented `editor.onCommandKey({}, 0, 27)` - which works fine for me.
